src/components/confirm-leave-mixin.cjsx

```diff
@@ -0,0 +1,12 @@
+$ = require 'jquery'
+
+module.exports =
+  # getFlux: -> {store, actions}
+  # getId: -> id 
+  componentDidMount: ->
+    $(window).on("beforeunload", @confirmLeave)
```

Comment:
I feel like it's not bad in cases when:
1. We're not manipulating Dom
2. jQuery is saving us from additional logic

I'm not sure what the best solution in this case is.  Is there a jquery-like library without dom manipulation?